wgpu/src/widget/canvas/layer/cache.rs

```diff
@@ -13,7 +13,7 @@ use std::{cell::RefCell, marker::PhantomData, sync::Arc};
 ///
 /// [`Layer`]: ../trait.Layer.html
 /// [`Cache`]: struct.Cache.html
-#[derive(Debug)]
+#[derive(Debug, Default)]
```

Comment:
I thinks this adds a `Default` bound to `T`, unnecessarily. It may be better to implement the trait manually.

Reply:
I thought this too and checked using cargo expand. The impl for default has the trait bound, it imposes no additional traits on the struct itself.

https://gist.github.com/robjtede/316a2cde764e28a78f24b73ef7477fcb

Reply:
Sorry, my point is that the `Default` implementation for `Cache` doesn't need a `Default` bound on `T`, as we do not need it to initialize the `PhantomData`.

![image](https://user-images.githubusercontent.com/518289/78612858-a2b72700-786a-11ea-8a2a-c088e0bf1742.png)

Reply:
Excellent idea.

<img width="330" alt="2020-04-08_23 11 35-7f0b1bc9" src="https://user-images.githubusercontent.com/3316789/78838552-54cb2c00-79ee-11ea-8c40-47a2b84a2372.png">


Updated the canvas examples too.